scrapy/spiders/__init__.py

```diff
@@ -18,8 +19,8 @@ class Spider(object_ref):
     class.
     """
 
-    name = None
-    custom_settings = None
+    name: Optional[str] = None
```

Comment:
This seems to fail in Python 3.5

Reply:
Indeed, it's not supported, that's why the PR is under the Python 3.6+ milestone.

Reply:
:facepalm:

Serves me right for reviewing draft pull requests.